### PR TITLE
test: that can not reuse code

### PR DESCRIPTION
**Pull request by danielgent**
~Wow, this test actually does fail~ :smile: 

~This was one flow that I expected to _"just work"_~ :sunglasses: 

I fixed this on https://github.com/sesamyab/auth/pull/803 Wasn't a real bug as such, just returning the wrong status code while returning an error message in the JSX template

### CODE REVIEW

File: test/integration/login/code-flow-liquidjs.spec.ts [hunk 1]
```diff
@@ -495,6 +495,68 @@ describe("Login with code on liquidjs template", () => {
...
+    const secondAuthenticateResponse = await oauthClient.u["enter-code"].$post({
+      query: {
+        state: enterCodeQuery.state,
+      },
+      form: {
+        code,
+      },
+    });
+
+    expect(secondAuthenticateResponse.status).toBe(400);
```
Comment (danielgent) [inline test/integration/login/code-flow-liquidjs.spec.ts:557]: Fails here because *we can use the same code multiple times*

Comment (danielgent) [reply test/integration/login/code-flow-liquidjs.spec.ts:557]: Ooooo, I hadn't thought about this

Because we're doing

```
  } catch (err) {
        console.error(err);
        return ctx.html(
          <LoginEnterCodePage
            vendorSettings={vendorSettings}
            email={session.authParams.username}
          />,
        );
      }
```

which is strange. As it's returning the same form again with a 200. It should be a 400 *and* it should have an error message. Probably a generic one like on login2 - the same if invalid or expired

Comment (danielgent) [reply test/integration/login/code-flow-liquidjs.spec.ts:557]: This will be fixed by https://github.com/sesamyab/auth/pull/802

File: test/integration/login/code-flow-liquidjs.spec.ts [hunk 2]
```diff
@@ -541,3 +603,6 @@ describe("Login with code on liquidjs template", () => {
     await snapshotResponse(incorrectCodeResponse);
   });
 });
+
+// TESTS TO COPY OVER
+// - edge cases block: probably this relies on the same validateCode() helper so isn't strictly necessary BUT would be a nice-to-have
```
Comment (danielgent) [inline test/integration/login/code-flow-liquidjs.spec.ts:608]: I wrote the same concern you raised in the PR about duplicating tests :+1: 

We can assume logic is shared and not test... but then we can break things and our tests won't pick them up as we're assuming internals :shrug:

**Review by markusahlstrand — APPROVED** [review]
This looks good, but it's a question if a 400 is the correct code? I would have guessed a 403 or a 401? Or maybe a 200 as it returns valid html.. What does auth0 do?
